### PR TITLE
Push notifications spec updates

### PR DESCRIPTION
**Pull request by paddybyers**
Draft spec changes for updates to the push device authentication mechanism. Also see https://github.com/ably/wiki/issues/243 for background.

### CODE REVIEW

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -989,6 +992,12 @@ h3(#activation-state-machine). Activation state machine
 * @(RSH4)@ When an event is fired and a transition from the current state is not defined for such event, the event is put into a queue. Then, whenever a transition happens, an event is dequeued from the queue. If a transition from the new current state is defined for the dequeued event, such transition happens. If not, the event is put back in its place in the queue. E. g. we're @WaitingForDeregistration@, and an event @CalledActivate@ happens. This event will be put in the queue, since there's no transition defined for it. Then, an event @Deregistered@ arrives, causing a transition to @NotActivated@. Now we peek the next item on the queue: @CalledActivate@. Because @NotActivated@ transitions on @CalledActivate@, the event is consumed and the machine transitions.
```
Comment (funkyboy) [inline content/client-lib-development-guide/features.textile]: @paddybyers This conflicts with https://github.com/ably/docs/pull/399/commits/ca19daac56e1f87132bd22410d12c643906c3212#diff-11900b395df266bc2bbfc1d11bdfc7a0R902 Does it belong to RSH3?

Comment (paddybyers) [reply content/client-lib-development-guide/features.textile]: What's the conflict?

**Comment by funkyboy** [comment]
There’s two items labeled RSH4

--
Cesare Rocchi
studiomagnolia.com

**Comment by funkyboy** [comment]
@paddybyers see also https://travis-ci.org/ably/docs/builds/357276108#L856

I think we should relabel the "Push channels" section as RSH7. Ok?

**Comment by paddybyers** [comment]
> I think we should relabel the "Push channels" section as RSH7. Ok?

sgtm

**Comment by mattheworiordan** [comment]
I have not reviewed this in detail, but how ready is this for others to implement?

**Comment by mattheworiordan** [comment]
Also, will we close https://github.com/ably/docs/pull/329 once this is merged?

**Comment by paddybyers** [comment]
> how ready is this for others to implement?

I will be making a few clarifications as a result of the iOS implementation, and then it's ready for the other libs to proceed.

**Pull request merged**